bezier-rs/lib/src/lib.rs

```diff
@@ -305,4 +305,190 @@ impl Bezier {
 		};
 		bezier_starting_at_t1.split(adjusted_t2)[t2_split_side]
 	}
+
+	// TODO: move this to a util file maybe
+	fn get_closest_point_in_lut(&self, lut: &Vec<DVec2>, point: DVec2) -> (f64, usize) {
+		let mut min_distance = f64::MAX;
+		let mut min_position = 0;
+		let mut dist;
+		for (idx, lut_point) in lut.iter().enumerate() {
```

Comment:
You could probably tighten this logic in this function by mapping to distance and then using min_by(). Something like:
`lut.map(|p| point.distance(*lut_point)).enumerate().min_by(|x| x.0)`
That being said I'm not sure if this is an appropriate style for rust.

Reply:
Thanks @RobNadal, really appreciate this!